File: src/main/java/no/nav/permitteringsskjemaapi/integrasjon/arbeidsgiver/ArbeidsgiverRapportLoggingProdusent.java
package no.nav.permitteringsskjemaapi.integrasjon.arbeidsgiver;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
import org.springframework.stereotype.Service;

@Service
@ConditionalOnMissingBean(PermitteringsskjemaProdusent.class)
public class ArbeidsgiverRapportLoggingProdusent implements Arbeidsgiver {

    private static final Logger LOG = LoggerFactory.getLogger(ArbeidsgiverRapportLoggingProdusent.class);
    private final String topic;

    public ArbeidsgiverRapportLoggingProdusent(@Value("${kafka.topic}") String topic) {
        this.topic = topic;
    }

    @Override
    public void publiser(ArbeidsgiverRapport rapport) {
        LOG.info("Sender {} på {}", rapport, topic);
    }
}

File: src/main/kotlin/no/nav/permitteringsskjemaapi/integrasjon/arbeidsgiver/ArbeidsgiverRapportLoggingProdusent.kt
package no.nav.permitteringsskjemaapi.integrasjon.arbeidsgiver

import no.nav.permitteringsskjemaapi.config.logger
import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
import org.springframework.stereotype.Service

@Service
@ConditionalOnMissingBean(PermitteringsskjemaProdusent::class)
class ArbeidsgiverRapportLoggingProdusent : Arbeidsgiver {
    private val topic = "permittering-og-nedbemanning.aapen-permittering-arbeidsgiver"
    private val log = logger()

    override fun publiser(rapport: ArbeidsgiverRapport?) {
        log.info("Sender {} på {}", rapport, topic)
    }
}